drivers/rtc/rtc_pcf8563.c

```diff
@@ -446,9 +415,41 @@ static int pcf8563_alarm_set_callback(const struct device *dev, uint16_t id,
 	gpio_add_callback(config->int1.port, &data->int1_callback);
 	LOG_DBG("Alarm set");
 	return 0;
-#endif
+#endif /* PCF8563_INT1_GPIOS_IN_USE */
 }
 
+#endif /* CONFIG_RTC_ALARM */
+
+#ifdef PCF8563_INT1_GPIOS_IN_USE
+/* The logic related to the pin interrupt logic */
+
+void callback_work_handler(struct k_work *work)
```

Comment:
please add `static` keyword to the functions as well, with this, the warning would likely have been caught faster :)